avro/src/main/java/com/fasterxml/jackson/dataformat/avro/deser/RecordReader.java

```diff
@@ -22,15 +22,18 @@ abstract class RecordReader extends AvroStructureReader
     protected final int _count;
 
     protected RecordReader(AvroReadContext parent, AvroFieldReader[] fieldReaders, AvroParserImpl parser, String typeId)
+            throws IOException
     {
         super(parent, TYPE_OBJECT, typeId);
         _fieldReaders = fieldReaders;
         _parser = parser;
         _count = fieldReaders.length;
+        if (parser != null)
```

Comment:
Can that be `null`? Maybe for some tests?

Reply:
a lot of tests seem to fail without this